File: codingame/community/theExperienceForCreatingPuzzles/theExperienceForCreatingPuzzles.rb
def toLvl (lvl)
    return (lvl * Math.sqrt(lvl) * 10).floor
end

lvl = gets.to_i
needed = gets.to_i
xp = gets.to_i * 300

while xp >= needed do
    lvl += 1
    xp -= needed
    needed = toLvl(lvl)
end

puts lvl
puts needed - xp

File: codingame/community/theExperienceForCreatingPuzzles/theExperienceForCreatingPuzzles.lua
lvl = tonumber(io.read())
needed = tonumber(io.read())
xp = tonumber(io.read()) * 300


while xp >= needed do
    lvl = lvl + 1
    xp = xp - needed
    needed = math.floor((lvl * math.sqrt(lvl) * 10))
end

print(lvl)
print(needed - xp)

